SwanCustomEnvironments/swancustomenvironments/templates/customenvs.html

```diff
@@ -0,0 +1,160 @@
+{% extends "page.html" %}
+
+{% block stylesheet %}
+{{super()}}
+<style type="text/css">
+/* disable initial hide */
+div#header, div#site {
+    display: block;
+}
+#output-box {
+    display: block;
+    width: 1150px;
+    height: 300px;
+    overflow-y: scroll;
+    overflow-x: hidden;
+    white-space: pre-wrap;
+    text-align: left;
+    border: 1px solid #ccc;
+    background-color: #f7f7f7;
+    padding: 10px;
+    border-radius: 5px;
+}
+#back_button {
+    display: none;
+    text-align: center;
+    background-color: #fc6404;
+    position: absolute;
+    left: 50%;
+    transform: translateX(-50%);
+    color: white;
+    border: none;
+    border-radius: 5px;
+    padding: 10px 20px;
+    font-size: 16px;
+    cursor: pointer;
+    transition: background-color 0.3s ease;
+}
+#next_button {
+    display: none;
+    text-align: center;
+    background-color: #0454a4;
+    position: absolute;
+    left: 50%;
+    transform: translateX(-50%);
+    color: white;
+    border: none;
+    border-radius: 5px;
+    padding: 10px 20px;
+    font-size: 16px;
+    cursor: pointer;
+    transition: background-color 0.3s ease;
+}
+</style>
+{% endblock %}
+
+{% block params %}
+{{super()}}
+data-base-url="{{base_url | urlencode}}"
+{% endblock %}
+
+{% block site %}
+
+<div class="container">
+    <div class="row">
+        <div class="swan-info">
+            <div id="swan-loader">
+                <div class="loader-circle">
+                    <img src="{{hub_prefix}}static/swan/logos/{{swan_logo_filename}}">
+                </div>
+                <div class="loader-line-mask">
+                    <div class="loader-line"></div>
+                </div>
+                <span class="text">Creating environment...</span>
+            </div>
+        </div>
+        <br><br>
+        <div id="output-box"></div>
+        <br>
+        <button id="back_button">Back</button>
+        <button id="next_button">Next</button>
+    </div>
+</div>
+
+{% endblock %}
+
+{% block script %}
+    {{super()}}
+
+<script type = "text/javascript" >
+    
+    var base_url = document.body.getAttribute('data-base-url');
+    
+    const urlParams = new URLSearchParams(window.location.search);
+    var env_name = urlParams.get('env') || "";
+    var repository = urlParams.get('repo') || "";
+    var accpy_version = urlParams.get('accpy') || "";
+    
+    var project_folder = '/';
+    if (repository.startsWith('http')) {
+        // Extract repository name from URL
+        const pattern = /^https?:\/\/[^/]+\/([^/\s]+)\/([^/\s]+)/;
+        const match = repository.match(pattern);
+        project_folder = match ? `/tree/SWAN_projects/${match[2]}` : '';
+    } else {
+        // Extract CERNBox path for EOS path repository, e.g /eos/user/u/username/project/local -> /tree/project/local
+        const pattern = /^\/eos\/user\/([^/]+)\/([^/]+)(\/.*)?$/;
```

Comment:
Same comment as above, this time for an EOS repo.

Mmm what I don't know is what happens here if the user specifies a non /eos/user path, @diocas ?

Reply:
I believe (and hope so) there's no way to redirect the users to a path outside of their home path.
That's why I thought I wouldn't be a good idea to allow a path with just /eos in the spawner.

I suppose the best thing to do was to make sure we got at least 4 elements in the EOS path `/eos`, `/user`, initial (`/u`) and `/username`.

This way we can be a little more confident that the user is accessing user content (not necessarily his, can be acessing resources of a mate)

Reply:
We need to match, for now, /eos/user (here and everywhere we match). Reason: the UI is not ready to access files not on eos user.